File: Server/src/main/core/network/amsc/ManagementServerState.java
package core.network.amsc;

import core.game.node.entity.player.Player;
import core.game.world.repository.Repository;
import core.network.packet.PacketRepository;
import core.network.packet.context.ContactContext;
import core.network.packet.outgoing.ContactPackets;

public enum ManagementServerState {

    NOT_AVAILABLE(2),

    CONNECTING(1),

    AVAILABLE(2);

    private final int value;

    private ManagementServerState(int value) {
        this.value = value;
    }

    public void set() {
        for (Player player : Repository.getPlayers()) {
            PacketRepository.send(ContactPackets.class, new ContactContext(player, ContactContext.UPDATE_STATE_TYPE));
        }
    }

    public int value() {
        return value;
    }
}

File: Emulator/src/main/core/net/amsc/ManagementServerState.kt
package core.net.amsc

import core.game.world.repository.Repository
import core.net.packet.PacketRepository
import core.net.packet.context.ContactContext
import core.net.packet.outgoing.ContactPackets

/**
 * The management server state.
 * @author Emperor
 */
enum class ManagementServerState(val value: Int) {

    /**
     * Represents the state when the management server is not available
     */
    NOT_AVAILABLE(2),

    /**
     * Represents the state when the management server is connecting
     */
    CONNECTING(1),

    /**
     * Represents the state when the management server is available
     */
    AVAILABLE(2);

    /**
     * Called when the state gets set.
     */
    fun set() {
        for (player in Repository.players) {
            PacketRepository.send(ContactPackets::class.java, ContactContext(player, ContactContext.UPDATE_STATE_TYPE))
        }
    }
}
